include/pybind11/gil.h

```diff
@@ -199,3 +199,43 @@ class gil_scoped_release {
 PYBIND11_NAMESPACE_END(PYBIND11_NAMESPACE)
 
 #endif // !PYBIND11_SIMPLE_GIL_MANAGEMENT
+
+PYBIND11_NAMESPACE_BEGIN(PYBIND11_NAMESPACE)
+
+class scoped_critical_section {
```

Comment:
Suggestion to move this to `gil_simple.h`, but with this name (`scoped_critical_section`).

Why: It will be usable without bringing in `#include  "detail/internals.h"`

Reply:
Maybe we should just make a new file. What's the status of gil_simple? Should we have changed the default? Don't remember if there were drawbacks. @tonybaloney was talking to me at PyCon about GIL management issues, I wonder if that would help.